tests/framework/clients/rancher/client.go

```diff
@@ -88,13 +95,47 @@ func clientOpts(restConfig *rest.Config, rancherConfig *Config) *clientbase.Clie
 	}
 }
 
-// AsUser accepts a user object, and then creates a token for said `user`. Then it instantiates and returns a Client using the token created
-func (c *Client) AsUser(user *management.User) (*Client, error) {
-	token := &management.Token{
-		UserID: user.ID,
+// DoAction is used to post an action to an endpoint, and marshal the response into the output parameter.
+func (c *Client) DoAction(endpoint, action string, body []byte, output interface{}) error {
+	url := "https://" + c.restConfig.Host + endpoint + "?action=" + action
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(body))
+	if err != nil {
+		return err
+	}
+
+	req.Header.Add("Authorization", "Bearer "+c.restConfig.BearerToken)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := c.Management.APIBaseClient.Ops.Client.Do(req)
```

Comment:
I just noticed this, I think after the request we should register a delete function, to make sure the token is deleted as well on cleanup.

Reply:
an alternative would be to make this a private method given we only use it for the token shenanigans.